File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l = 0;
    let r = heights.length - 1;
    let maxArea = 0;
    // min(heght1, height2) * width 
    while (l < r) {
        let width = (r - l)
        maxArea = Math.max(Math.min(heights[l], heights[r]) * width, maxArea);
        if (heights[l] > heights[r]) {
            r--;
        }
        else if (heights[r] > heights[l]) {
            l++;
        }
        else {
            l++;
        }
    }
        return maxArea;
    }

}

File: Data Structures & Algorithms/pow-x-n/submission-0.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0 :
            return 1
        if n % 2 == 0:
            if n > 0:
                return self.myPow(x ,n //2 ) * self.myPow(x , n//2)
            if n < 0:
                return 1/(self.myPow(x , -n //2 ) * self.myPow(x , -n//2))
        else:
            if n > 0:
                return  self.myPow(x ,n //2 ) * self.myPow(x , n//2) * x
            else:
                return 1/(self.myPow(x ,-n //2 ) * self.myPow(x , -n//2) * x)